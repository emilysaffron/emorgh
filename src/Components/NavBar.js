import ArticlePage from "../Pages/ArticlePage";
import React, { useState } from "react";
import Home from "../Pages/Home";
import { Router, Link } from "@reach/router";

import { ScrollableNavigation } from "@bbc/psammead-navigation/scrollable";
import Navigation, {
  NavigationUl,
  NavigationLi,
} from "@bbc/psammead-navigation";

import Rate from "../Pages/Rate";
import Submitted from "../Pages/Submitted";

function NavBar() {
  // Maybe be more explicit here that pageNo is 'currentPageNumber'
  // no === 'number'
  // number === 'number'   DONE

  const [currentPageNumber, updatePage] = useState(0);

  const loadNextArticle = () => {
    if (currentPageNumber < 5) {
      updatePage(currentPageNumber + 1);
    }
  };

  const loadPreviousArticle = () => {
    if (currentPageNumber > 1) {
      updatePage(currentPageNumber - 1);
    }
  };

  return (
    <div className="Nav">
      <Navigation>
        <ScrollableNavigation>
          <NavigationUl>
            <Link to={"/"}>
              <NavigationLi>Home</NavigationLi>
            </Link>
            {/* Next/Previous links - can this be improved by passing a single 'PageNav' component some props? */}
            {/* Can the currentPageNumber calc be handled in its own function to remove the ternary here? */}
            {currentPageNumber < 5 ? (
              <Link to={`/article${currentPageNumber + 1}`}>
                <NavigationLi onClick={loadNextArticle}>Next</NavigationLi>
              </Link>
            ) : (
              <Link to={`/article${currentPageNumber}`}>
                <NavigationLi onClick={loadNextArticle}>Next</NavigationLi>
              </Link>
            )}
            {currentPageNumber > 1 ? (
              <Link to={`/article${currentPageNumber - 1}`}>
                <NavigationLi onClick={loadPreviousArticle}>
                  Previous
                </NavigationLi>
              </Link>
            ) : (
              <Link to={`/article${currentPageNumber}`}>
                <NavigationLi onClick={loadPreviousArticle}>
                  Previous
                </NavigationLi>
              </Link>
            )}

            <Link to={"/rate"}>
              <NavigationLi>Rate the Articles</NavigationLi>
            </Link>
          </NavigationUl>
        </ScrollableNavigation>
      </Navigation>

      {/* These are not part of the nav bar- they should be within App. Also - what about creating a separate 'Pages' 
      folder that holds each of these page-level components below, to distinguish them from components */}
      <Router>
        <ArticlePage
          exact
          path={`/article${currentPageNumber}`}
          num={currentPageNumber}
        />
      </Router>
      <Router>
        <Rate exact path={"/rate"} />
      </Router>

      <Router>
        <Home exact path={"/"} />
      </Router>
      <Router>
        <Submitted exact path={"/submitted"} />
      </Router>
    </div>
  );
}

export default NavBar;
