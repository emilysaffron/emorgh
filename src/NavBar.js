import ArticlePage from "./ArticlePage";
import React, { useState } from "react";
import Home from "./Home";
import { Router, Link } from "@reach/router";

import { ScrollableNavigation } from "@bbc/psammead-navigation/scrollable";
import Navigation, {
  NavigationUl,
  NavigationLi,
} from "@bbc/psammead-navigation";

import Rate from "./Rate";
import Submitted from "./Submitted";

function NavBar() {
  // Maybe be more explicit here that pageNo is 'currentPageNumber'
  // no === 'number'
  // number === 'number'
  const [pageNo, updatePage] = useState(0);
  // What is 'rate'? Why is it within 'NavBar'?
  // this 'rate' has the same name as a component 'Rate'
  const [rate, updateRate] = useState(false);
  const loadNextArticle = () => {
    if (pageNo < 5) {
      updatePage(pageNo + 1);
    }
  };

  const loadPreviousArticle = () => {
    if (pageNo > 1 && rate === false) {
      updatePage(pageNo - 1);
    }
    updateRate(false);
  };

  const toggleRate = () => {
    updateRate(true);
  };

  return (
    <div className="Nav">
      <Navigation>
        <ScrollableNavigation>
          <NavigationUl>
            <Link to={"/"}>
              <NavigationLi>Home</NavigationLi>
            </Link>
            {/* Next/Previous links - can this be improved by passing a single 'PageNav' component some props? */}
            {/* Can the pageNo calc be handled in its own function to remove the ternary here? */}
            {pageNo < 5 ? (
              <Link to={`/article${pageNo + 1}`}>
                <NavigationLi onClick={loadNextArticle}>Next</NavigationLi>
              </Link>
            ) : (
              <Link to={`/article${pageNo}`}>
                <NavigationLi onClick={loadNextArticle}>Next</NavigationLi>
              </Link>
            )}
            {pageNo > 1 ? (
              <Link to={`/article${pageNo - 1}`}>
                <NavigationLi onClick={loadPreviousArticle}>
                  Previous
                </NavigationLi>
              </Link>
            ) : (
              <Link to={`/article${pageNo}`}>
                <NavigationLi onClick={loadPreviousArticle}>
                  Previous
                </NavigationLi>
              </Link>
            )}

            <Link to={"/rate"} onClick={toggleRate}>
              <NavigationLi>Rate the Articles</NavigationLi>
            </Link>
          </NavigationUl>
        </ScrollableNavigation>
      </Navigation>

      {/* These are not part of the nav bar- they should be within App. Also - what about creating a separate 'Pages' 
      folder that holds each of these page-level components below, to distinguish them from components */}
      <Router>
        <ArticlePage exact path={`/article${pageNo}`} num={pageNo} />
      </Router>
      <Router>
        <Rate exact path={"/rate"} rate={rate} />
      </Router>

      <Router>
        <Home exact path={"/"} />
      </Router>
      <Router>
        <Submitted exact path={"/submitted"} rate={rate} />
      </Router>
    </div>
  );
}

export default NavBar;
