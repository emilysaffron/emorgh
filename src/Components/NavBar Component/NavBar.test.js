import NavBar from "./NavBar";
import renderer from "react-test-renderer";

it("NavBar Snapshot", () => {
    const tree = renderer.create(<NavBar  />).toJSON();
    expect(tree).toMatchSnapshot();
  });