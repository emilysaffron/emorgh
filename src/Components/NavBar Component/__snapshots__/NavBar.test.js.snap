// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Snapshot 1`] = `
<div
  className="Nav"
>
  <nav
    className="css-5vwsm5-StyledNav e1lim4kn5"
    dir="ltr"
    role="navigation"
  >
    <div
      className="css-h6pr6t-NavWrapper e1lim4kn0"
    >
      <div
        className="css-f0nofw-StyledScrollableNav e1fc18nj0"
        dir="ltr"
      >
        <ul
          className="css-1p24brr-StyledUnorderedList e1lim4kn1"
          role="list"
        >
          <a
            href="/articleundefined"
            onClick={[Function]}
          >
            <li
              className="css-1vectme-StyledListItem e1lim4kn3"
              dir="ltr"
              role="listitem"
            >
              <a
                className="css-cig9wk-StyledLink e1lim4kn2"
                onClick={[Function]}
              >
                Previous
              </a>
            </li>
          </a>
          <a
            href="/articleundefined"
            onClick={[Function]}
          >
            <li
              className="css-1vectme-StyledListItem e1lim4kn3"
              dir="ltr"
              role="listitem"
            >
              <a
                className="css-cig9wk-StyledLink e1lim4kn2"
                onClick={[Function]}
              >
                Next
              </a>
            </li>
          </a>
          <a
            href="/rate"
            onClick={[Function]}
          >
            <li
              className="css-1vectme-StyledListItem e1lim4kn3"
              dir="ltr"
              role="listitem"
            >
              <a
                className="css-cig9wk-StyledLink e1lim4kn2"
              >
                Rate the Articles
              </a>
            </li>
          </a>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
